### PR TITLE
Offer more details to users about the ownership check on FAT32

### PR DESCRIPTION
**Pull request by dscho**
Since FAT cannot store ownership information, users cannot do anything about "unsafe ownership" there. Except exempt the directory via the `safe.directory` mechanism. Inform the users about this (rather than the bogus "World owns this" message).

Since we cannot know at the time the ownership check is performed whether Git will require a valid repository or not, we keep mum in the general case, and only say something if the `GIT_TEST_DEBUG_UNSAFE_DIRECTORIES` knob is on.

Speaking of that knob, we now hint to the users that they can use this knob to find out more, in case that the ownership fails (and Git absolutely requires a valid repository).

This addresses https://github.com/git-for-windows/git/issues/3886

### CODE REVIEW

**Comment by rimrul** [comment]
This feels like reintroducing the vulnerability that the whole ownership check is supposed to fix.

Are you sure this shouldn't at least produce a big warning?

**Comment by dscho** [comment]
Hmm. You're right: just because the user can't fix the issue doesn't mean that they're not vulnerable.

Let me mull this over for a bit.

**Comment by dscho** [comment]
> Let me mull this over for a bit.

So I mulled over this and agree with @rimrul that we cannot relax the ownership check. If the user runs Git on a file system that does not record ownership information, the safest option is to require an explicit exception (via that `safe.directory` config setting). We _can_ be more helpful, though, and this is what this PR now does.

File: compat/mingw.c [hunk 1]
```diff
@@ -3541,7 +3557,14 @@ int is_path_owned_by_current_sid(const char *path)
...
 			 */
 			result = 1;
-		else if (git_env_bool("GIT_TEST_DEBUG_UNSAFE_DIRECTORIES", 0)) {
+		else if (IsWellKnownSid(sid, WinWorldSid) &&
+			 git_env_bool("GIT_TEST_DEBUG_UNSAFE_DIRECTORIES", 0) &&
+			 !acls_supported(path)) {
+			/*
+			 * On FAT32 volumes, ownership is not actually recorded.
+			 */
+			warning("'%s' is on a file system that cannot represent ownership", path);
```
Comment (dscho) [inline compat/mingw.c:3566]: @derrickstolee do you think "that does not record ownership" would be more accurate/elegant?

Comment (derrickstolee) [reply compat/mingw.c:3566]: I agree that "does not record" is the most accurate. I read "cannot represent" as "cannot tell me" which is different than "doesn't even keep track of".

**Review by derrickstolee — APPROVED** [review]
Do we have any CI that tests FAT32? How did you validate this on that platform?

**Pull request merged**